### PR TITLE
feat(#19): add feed detail endpoint

### LINKED ISSUE #19: Update Details Endpoint
`/api/feed/<neighbourhoodId>`

determine the timepast field..
check formulation for finalRank output..

```json
{ 
  "neighbourhoodId": "<-id->", 
  "lastUpdateTime": "<-timepast?->", 
  "intensity": "<-finalRank->", 
  "details": [ 
    "<-building-name-0-> - <-ballot_boxes.box_no in that building_id-0->",
    "<-building-name-1-> - <-ballot_boxes.box_no in that building_id-1->"
  ], 
}
```

### PR DESCRIPTION
**Pull request by salihdev0**
## What was done ?

- Feed detail endpoint has been created.
- Added repository function to feed this endpoint.
- Removed unique index for `location_id` in `volunteer_counts` table. Because there can be more than one building in a neighborhood.

## Notes

- `lastUpdateTime` data was requested, but we do not keep any time data in the database.

- I honestly don't know how to generate the `intensity` data.

## Endpoint

URL: `/feed/<neighborhood_id>`

What kind of response is coming?

```json
{
    "neighborhoodId": 1,
    "lastUpdateTime": "2023-05-24T11:37:09Z",
    "intensity": 1,
    "details": [
        {
            "buildingName": "deneme",
            "ballotBoxNos": [
                123123,
                12312312
            ]
        },
        {
            "buildingName": "test",
            "ballotBoxNos": [
                343
            ]
        }
    ]
}
```

## Query

The SQL query this endpoint uses:

```sql
WITH selected_neighbourhood AS (
  SELECT id
  FROM locations
  WHERE neighbourhood_id = $1
  LIMIT 1
), box_numbers AS (
  SELECT b.id AS building_id, ARRAY_AGG(bb.box_no ORDER BY bb.box_no) AS box_numbers
  FROM volunteer_counts vc
  JOIN selected_neighbourhood sn ON vc.location_id = sn.id
  LEFT JOIN buildings b ON vc.building_id = b.id
  LEFT JOIN ballot_boxes bb ON b.id = bb.building_id
  GROUP BY b.id
)
SELECT b.name AS building_name, bn.box_numbers
FROM buildings b
LEFT JOIN box_numbers bn ON b.id = bn.building_id;
```

Closes #19

### CODE REVIEW

File: resources/init.sql
Comment (gurkanguray) [inline resources/init.sql:1]: We should add `neighbourhood_id` as a unique column to the `locations` table. The `neighbourhood_id` that frontend sends does not refers to the `neighbourhood_ysk_id` These both have different values for the same `location_id`

File: resources/init.sql [hunk 1]
```diff
@@ -78,7 +78,7 @@ CREATE TABLE "volunteer_counts" (
     "priority" int,
     "source_id" int REFERENCES "sources" ("id"),
     "building_id" int REFERENCES "buildings" ("id"),
-    "location_id" int unique REFERENCES "locations" ("id"),
+    "location_id" int REFERENCES "locations" ("id"),
     "neighbourhood_id" int unique
```
Comment (gurkanguray) [inline resources/init.sql]: You can think of `location_id` as a `neighbourhood_id`. They have very similar purposes but with different values. Therefore the unique statement of `neighbourhood_id` should also be removed.

File: repository/repository.go [hunk 2]
```diff
@@ -73,6 +75,47 @@ func (r *Repository) Close() {
...
 }
 
+func (r *Repository) GetFeedDetail(neighborhoodId int) (*feeds.FeedDetailResponse, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+	defer cancel()
+	sql := `
+	WITH selected_neighbourhood AS (
+		SELECT id
+		FROM locations
+		WHERE neighbourhood_ysk_id = $1
```
Comment (gurkanguray) [inline repository/repository.go]: The request will send `neighbourhood_id`, which is different than the `neighbourhood_ysk_id.` Please see my comment below

File: repository/repository.go
Comment (gurkanguray) [inline repository/repository.go:1]: If the `details` field in the response changes, we should convert the result of the `string_agg` to the list of int

**Review by gurkanguray — CHANGES_REQUESTED** [review]
Please note that now the `details` field structure has changed. Great work! Thank you for your efforts...

File: repository/repository.go (hunk 2 above)
Comment (salihdev0) [reply repository/repository.go]: we don't have the `neighbourhood_id` with the v3 schema of the database (in #16). Do I add this field again and deal with it?

https://user-images.githubusercontent.com/6787068/240446313-5055a072-3965-4391-99ae-58635a40050a.jpeg

Comment (gurkanguray) [reply repository/repository.go]: yes, please

File: repository/mock/feeds.go [hunk 3]
```diff
@@ -21,3 +24,28 @@ func GetFeeds() (*feeds.Response, error) {
...
+		if nh.Id == neighborhoodId {
+			response.NeighborhoodId = nh.Id
+			response.Intensity = rand.Intn(5) + 1
+			response.LastUpdateTime = time.Now().Add(-time.Minute * time.Duration(rand.Intn(60))).Format(time.RFC3339)
+			response.Details = make([]feeds.FeedDetail, 0)
+			for i := 0; i < rand.Intn(10)+1; i++ {
+				response.Details = append(response.Details, feeds.FeedDetail{
+					BuildingName: fmt.Sprintf("Building %d", i),
+					BallotBoxNos: []int{
+						i - 1,
```
Comment (gurkanguray) [inline repository/mock/feeds.go]: The ballot box number shouldn't contain a negative value in the mock data. Mock data and the prod data should be consistent

**Review by gurkanguray — APPROVED** [review]

**Pull request merged**